### PR TITLE
Repointed all routes using cmpdreg web controller routes at new acas/api/v1 routes

### LINKED ISSUE #937: Repoint used node routes which use persistence  web routes to new /api/v1 routes
We are in the process of removing all web controllers and this ticket is point acas node routes at the new persistence web routes.

### PR DESCRIPTION
**Pull request by brianbolt**
## Description
Repointed all routes using cmpdreg web controller routes at new acas/api/v1 routes

## Related Issue
Fixes #937 
mcneilco/acas-roo-server#309

## How Has This Been Tested?
See testing on mcneilco/acas-roo-server#309

### CODE REVIEW

File: modules/CmpdReg/src/client/src/Parent.js [hunk 1]
```diff
@@ -156,7 +156,7 @@ $(function() {
 	            var optionToInsert = null;
             }
             this.stereoCategoryCodeController =
-                this.setupCodeController('stereoCategoryCode', 'stereoCategorys', 'stereoCategory', optionToInsert);
+                this.setupCodeController('stereoCategoryCode', 'stereoCategories', 'stereoCategory', optionToInsert);
```
Comment (brianbolt) [inline modules/CmpdReg/src/client/src/Parent.js:159]: Ended up with a different name here because we already had a roo `api/v1/stereoCategories` route.

File: modules/CmpdReg/src/client/src/AddAlias.js [hunk 2]
```diff
@@ -104,7 +104,7 @@
     AddAliasController.prototype.initialize = function() {
       this.collection = this.options.collection;
       this.listOfAliases = new AliasListList();
-      this.listOfAliases.type = "parentaliaskinds";
+      this.listOfAliases.type = "aliases/parentAliasKinds";
```
Comment (brianbolt) [inline modules/CmpdReg/src/client/src/AddAlias.js:107]: These were already setup in the roo /api/v1 routes via `aliases/parentAliasKinds`. Kind of awkward but type = the url this controller calls.

File: modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee [hunk 3]
```diff
@@ -104,40 +102,11 @@ exports.cmpdRegIndex = (req, res) ->
 			deployMode: global.deployMode
 			cmpdRegConfig: config.all.client.cmpdreg
 
-syncCmpdRegUser = (req, cmpdRegUser) ->
```
Comment (brianbolt) [inline modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee:107]: This was no longer being used (or rather it was being called but erroring).

File: modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee [hunk 4]
```diff
@@ -104,40 +102,11 @@ exports.cmpdRegIndex = (req, res) ->
...
-	cmpdRegCall = config.all.client.service.cmpdReg.persistence.basepath + "/" +endOfUrl
-	console.log cmpdRegCall
-	req.pipe(request(cmpdRegCall)).pipe(resp)
-
 exports.getAPICmpdReg = (req, resp) ->
 	console.log 'in getAPICmpdReg'
 	console.log req.originalUrl
 	endOfUrl = (req.originalUrl).replace /\/cmpdreg\//, ""
-	cmpdRegCall = config.all.client.service.cmpdReg.persistence.basepath + "/api/v1/" +endOfUrl
+	cmpdRegCall = config.all.client.service.cmpdReg.persistence.fullpath + "/" +endOfUrl
```
Comment (brianbolt) [inline modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee:109]: Figured I might as well use the full config path here.

File: modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee [hunk 5]
```diff
@@ -147,47 +116,6 @@ exports.getAuthorizedCmpdRegProjects = (req, resp) ->
 		resp.status "200"
 		resp.end JSON.stringify allowedUserProjects
 
-exports.saveProjects = (jsonBody, callback) ->
```
Comment (brianbolt) [inline modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee:150]: No longer being used since we migrated all of these to thing projects or passthrough.

File: modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee [hunk 6]
```diff
@@ -201,46 +129,6 @@ exports.getScientistsInternal = (callback) ->
 	loginRoutes.getAuthorsInternal {additionalCodeType: 'compound', additionalCodeKind: 'scientist', roleName: roleName}, (statusCode, authors) =>
 		callback authors
 
-exports.saveScientists = (jsonBody, callback) ->
```
Comment (brianbolt) [inline modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee:204]: Scientist routes no longer existed.

File: modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee [hunk 7]
```diff
@@ -478,29 +366,6 @@ exports.saveIsotopes = (req, resp) ->
 			resp.end JSON.stringify {error: "something went wrong :("}
 	)
 
-exports.molConvert = (req, resp) ->
```
Comment (brianbolt) [inline modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee:481]: I didn't see this being used anywhere.

**Comment by brianbolt** [comment]
I have the PR here set to merge into 1.13.7 but this shouldn't be merged until https://github.com/mcneilco/acas-roo-server/pull/309 is merged

File: modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee [hunk 8]
```diff
@@ -32,7 +32,6 @@ exports.setupRoutes = (app, loginRoutes) ->
 	app.post '/cmpdReg/metalots', loginRoutes.ensureAuthenticated, exports.metaLots
 	app.post '/cmpdReg/salts', loginRoutes.ensureAuthenticated, exports.saveSalts
 	app.post '/cmpdReg/isotopes', loginRoutes.ensureAuthenticated, exports.saveIsotopes
-	app.post '/cmpdReg/api/v1/structureServices/molconvert', loginRoutes.ensureAuthenticated, exports.molConvert
```
Comment (bffrost) [inline modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee:35]: Why was this removed? Thought this was used by MarvinJS

**Review by bffrost — COMMENTED** [review]
One line I had questions about, otherwise LGTM

File: modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee (hunk 8 above)
Comment (brianbolt) [reply modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee]: Thanks, I misread how this was being used.  I just added it back.

**Review by bffrost — APPROVED** [review]
LGTM
